airsonic-main/src/test/java/org/airsonic/player/dao/PodcastDaoTestCase.java

```diff
@@ -134,10 +134,10 @@ public void testGetEpisodes() {
 
         List<PodcastEpisode> episodes = podcastDao.getEpisodes(channelId);
         assertEquals("Error in getEpisodes().", 4, episodes.size());
-        assertEpisodeEquals(a, episodes.get(0));
-        assertEpisodeEquals(c, episodes.get(1));
-        assertEpisodeEquals(b, episodes.get(2));
-        assertEpisodeEquals(d, episodes.get(3));
+        assertEpisodeEquals(d, episodes.get(0));
+        assertEpisodeEquals(a, episodes.get(1));
+        assertEpisodeEquals(c, episodes.get(2));
+        assertEpisodeEquals(b, episodes.get(3));
```

Comment:
Hmmm, is this because null values are being treated higher order than populated now in the new db?

This will actually be a nuisance for people with podcasts, where all the null values will crowd the top instead of getting the newly imported ones first. People will never see the fresh podcasts!

I suggest we add `sql.nulls_first=false` to the jdbc connection String to retain the old behavior

so your connection string will be something like: `"return jdbc:hsqldb:file:" + getAirsonicHome().getPath() + "/db/" + getFileSystemAppName() + ";sql.enforce_size=false;sql.nulls_first=false";`

Documented here: http://hsqldb.org/doc/2.0/guide/dbproperties-chapt.html

Reply:
I don't mind doing that, but will other databases (PostgreSQL/MariaDB) do the same as HSQL? Shouldn't this be handled in another layer in this case?

Reply:
You're not using Postgres for the default db. This string is exclusively for our internal db and is thus specific to hsql db. It's in the string itself `return "jdbc:hsqldb:file:"`. I think we should keep it consistent with what the original hsql db was doing.

People with Postgres can provide their own connection parameters for when they connect to Postgres, that's not under our control.

As far as the general behavior of nulls, yes we should be using NULLS LAST in the order by clauses for our queries. But that's probably a separate PR. For now, I think we should keep our embedded DB consistent with what it was doing before, and keep behavior in line with what we expect to be the best (nulls last)